pyclaragenomics/claragenomics/utilities/evaluate_paf.py

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+#
+"""Functions and tools for calculating the accuracy of overlap detection"""
+
+import argparse
+
+
+def evaluate_paf(truth_paf_filepath, test_paf_filepath, pos_tolerance=500):
+    """Given a truth and test set PAF file, count number of in/correctly detected, and non-detected overlaps
+    Args:
+       truth_paf_filepath (str): Path to truth set PAF file
+       test_paf_filepath (str): Path to test set PAF file
+       pos_tolerance (int): query and referennce positions within this range will be connsidered to be a matched overlap
+
+    Returns: 3-tupe consisting of (rue_positive_count, false_positive_count, false_negative_count).
+    """
+    with open(truth_paf_filepath) as f:
+        truth_paf = f.readlines()
+
+    with open(test_paf_filepath) as f:
+        test_paf = f.readlines()
+
+    # Put the truth paf into a dictionary:
+    truth_overlaps = {}
+
+    for paf_entry in truth_paf:
+        overlap = paf_entry.split('\t')
+        key = overlap[0] + overlap[5]  # Unique key for this overlap
+        truth_overlaps[key] = (overlap[2], overlap[3], overlap[7], overlap[8])
+
+    true_positive_count = 0
+    false_positive_count = 0
+    false_negative_count = 0
+
+    for paf_entry in test_paf:
+        overlap = paf_entry.split('\t')
+
+        query_start_0 = int(overlap[2])
+        query_end_0 = int(overlap[3])
+        target_start_0 = int(overlap[7])
+        target_end_0 = int(overlap[8])
+
+        key = overlap[0] + overlap[5]  # Unique key for this overlap
+        key_reversed = overlap[5] + overlap[0]  # Unique key for this overlap, switching query and target
+
+        matched = False
+        if key in truth_overlaps:
+            truth_overlap = truth_overlaps[key]
+
+            query_start_1 = int(truth_overlap[0])
+            query_end_1 = int(truth_overlap[1])
+            target_start_1 = int(truth_overlap[2])
+            target_end_1 = int(truth_overlap[3])
+
+        elif key_reversed in truth_overlaps:
+            truth_overlap = truth_overlaps[key_reversed]
+
+            query_start_1 = int(truth_overlap[2])
+            query_end_1 = int(truth_overlap[3])
+            target_start_1 = int(truth_overlap[0])
+            target_end_1 = int(truth_overlap[1])
+
+        matched = abs(query_start_0 - query_start_1) < pos_tolerance and \
```

Comment:
if there are overlaps in the test paf that aren't there in the truth paf, then the `*_1` variables won't be defined for this computation right?

Reply:
Yes. Thanks for spotting this. Now fixed.